src/Platforms/AbstractPlatform.php

```diff
@@ -3882,6 +3896,28 @@ public function getTimeTypeDeclarationSQL(array $column)
         throw Exception::notSupported(__METHOD__);
     }
 
+    /**
+     * Adds a parameter to a TIME-type SQL declaration to indicate precision
+     *
+     * @param string  $decl   The SQL declaration to update e.g. TIME, DATETIME
+     * @param mixed[] $column The column definition
+     */
+    protected function addVariablePrecisionTimeSQL(string $decl, array $column): string
+    {
+        if (
+            ! isset($column['length']) ||
+            ! is_numeric($column['length']) ||
+            (int) $column['length'] < 0 ||
+            ((int) $column['length'] === 0 && $this->ignoreZeroTimePrecision())
+        ) {
+            return $decl;
+        }
+
+        $precision = min($this->getTimeMaxPrecision(), (int) $column['length']);
+
+        return preg_replace('/(\w*TIME\w*)(\(\d+\))?/', sprintf('$1(%d)', $precision), $decl, 1) ?? $decl;
```

Comment:
Parsing the string that the class itself generates looks like a hack. Parsing SQL with regular expressions is also a hack. Why is this necessary?

Reply:
Methodology revised. SQL is now generated by the platform without this hack.